src/core/server/deprecations/README.mdx

```diff
@@ -0,0 +1,194 @@
+
+All the config deprecation functions (`unused`, `unusedFromRoot`, `rename`, `renameFromRoot`) now accept an optional parameter to customize the deprecation details.
+
+### Example
+
+```ts
+export const config: PluginConfigDescriptor<ConfigType> = {
+  schema: configSchema,
+  deprecations: ({ renameFromRoot }) => [
+    renameFromRoot('ui_metric.debug', 'usageCollection.uiCounters.debug', {
+      documentationUrl: 'elastic.co/some-url',
+    }),
+  ],
+};
+```
+
+If the above deprecated kibana config `ui_metric.debug` is being used by the cluster.
+The service will show the following details about the deprecation:
+
+```ts
+{
+  "deprecationsInfo":[{
+    "level":"critical",
+    "message": "\"ui_metric.debug\" is deprecated and has been replaced by \"usageCollection.uiCounters.debug\"",
+    "documentationUrl": 'elastic.co/some-url',
+    "correctiveActions":{
+      "manualSteps": [
+        `Replace "ui_metric.debug" with "usageCollection.uiCounters.debug" in the Kibana config file, CLI flag, or environment variable (in Docker only).`,
+      },
+    "domainId":"usageCollection",
+  }],
+}
+```
+
+#### Custom deprecations for plugin configs
+Custom config deprecation handling allows specifying the deprecation details via the `addDeprecation`.
+
+##### Example:
+
+```ts
+export const config: PluginConfigDescriptor<ConfigSchema> = {
+  exposeToBrowser: {
+    defaultAppId: true,
+  },
+  schema: configSchema,
+  deprecations: () => [
+    (
+      completeConfig: Record<string, any>,
+      rootPath: string,
+      addDeprecation: AddConfigDeprecation
+    ) => {
+      if (
+        get(completeConfig, 'kibana.defaultAppId') === undefined &&
+        get(completeConfig, 'kibana_legacy.defaultAppId') === undefined
+      ) {
+        return completeConfig;
+      }
+      addDeprecation({
+        message: `kibana.defaultAppId is deprecated and will be removed in 8.0. Please use the \`defaultRoute\` advanced setting instead`,
+        correctiveActions: {
+          manualSteps: [
+            'Go to Stack Management > Advanced Settings',
+            'Update the "defaultRoute" setting under the General section',
+            'Remove "kibana.defaultAppId" from the kibana.yml config file',
+          ],
+        },
+      });
+      return completeConfig;
+    },
+  ],
+};
+```
+
+## How do I use this service for my deprecated plugin features?
+Plugins are responsible for registering any deprecations during the `setup` lifecycle by using the deprecations service.
+
+```ts
+coreSetup.deprecations.registerDeprecations({
+  getDeprecations: ({ esClient, savedObjectsClient }) => [{ ...`<list of deprecations>` }],
+});
+```
+
+The `getDeprecations` function is invoked when the user requests to see the deprecations affecting their deployment. The function is passed a context object `{ esClient, savedObjectsClient }`.
+
+```ts
+interface GetDeprecationsContext {
```

Comment:
Is it possible for us to link to the docs generated from code here instead of copying this in? That way we don't have to keep two things in sync.